src/getDataset.js

```diff
@@ -148,7 +151,7 @@ const getDataset = async (req, res) => {
     try {
       await requestCertainFileType(res, req, fetchUrls.additional, query);
     } catch (err) {
-      return helpers.handleError(res, `Couldn't fetch JSON: ${fetchUrls.additional}`, err.message);
+      return helpers.handle500Error(res, `Couldn't fetch JSON: ${fetchUrls.additional}`, err.message);
```

Comment:
For narratives, you now return `404` if the file doesn't exist. This should ideally be the case here as well.

Reply:
Or maybe 204 is more appropriate in this case? I'm not an expert on what's best here

Reply:
Hmm, I'm not sure either. I've been assuming 404, but after looking up 204, I can see either one as acceptable. Do you have any thoughts here, @tsibley ?

Reply:
In the meantime, I made it a 404 error.

Reply:
The code you've added doesn't work as expected because `requestCertainFileType` throws if the request 404s so we jump ahead to the `catch` block and therefore skip your `if (response.status === 404) {` check. You can test this by going to http://localhost:5000/zika which will attempt to fetch http://data.nextstrain.org/zika_root-sequence.json (which doesn't exist).

I think you'll have to change `utils -> fetchJSON()` to throw a custom `Error` (or similar) upon 404, and then check the `err` object in this catch block.

I tend to think this should be a 204, as we designed things around a "make lots of fetches and some of them are allowed to fail", but @tsibley will know the best code to use here.

Reply:
P.S. https://github.com/nextstrain/auspice/pull/1226/files is an example of using custom `Error`s. (Comments on that PR welcome!)

Reply:
Ok @jameshadfield , I'm finally getting back to this.

I rewrote the 404 errors as 204 errors and created a custom Error class like in the example PR from Auspice you linked.

I also did some extra error handling for unauthorized (403) errors, rewriting them as 404 errors to the client rather than 500. 

Please see the final two commits for these changes. I would appreciate any feedback, especially since I haven't been thinking in the Nextstrain-space for the past few weeks.

Reply:
See [my comment further down](https://github.com/nextstrain/nextstrain.org/pull/228#issuecomment-745713851) for the conclusion on this thread re: 404 vs. 204.